crates/factors-executor/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use spin_app::{App, AppComponent};
+use spin_core::Component;
+use spin_factors::{AsInstanceState, ConfiguredApp, RuntimeFactors, RuntimeFactorsInstanceState};
+
+/// A FactorsExecutor manages execution of a Spin app.
+///
+/// `Factors` is the executor's [`RuntimeFactors`]. `ExecutorInstanceState`
+/// holds any other per-instance state needed by the caller.
+pub struct FactorsExecutor<Factors: RuntimeFactors, ExecutorInstanceState = ()> {
+    factors: Factors,
+    core_engine: spin_core::Engine<InstanceState<Factors::InstanceState, ExecutorInstanceState>>,
+    configured_app: ConfiguredApp<Factors>,
+    component_instance_pres: HashMap<String, InstancePre<Factors, ExecutorInstanceState>>,
+}
+
+/// A ComponentLoader is responsible for loading Wasmtime [`Component`]s.
+pub trait ComponentLoader {
+    /// Loads a [`Component`] for the given [`AppComponent`].
+    fn load_component(&mut self, component: &AppComponent) -> anyhow::Result<Component>;
+}
```

Comment:
Nit: can we move this somewhere down below. I think it's more readable to have the `FactorsExecutor` impl block next to the struct definition.

Reply:
done